test/html.test.js

```diff
@@ -89,6 +89,117 @@ describe('html', function() {
       html.patch(elem, jml);
       assert.equal(elem.outerHTML, '<div id="a1"></div>');
     });
+
+    it('should update an element given JsonML with raw (Markup) instance', function() {
+      const rawStr = new html.raw('foo');
+      assert.equal(rawStr.value, 'foo');
+      assert.equal(rawStr.toString(), 'foo');
+      const elem = document.createElement('div');
+      const jml = ['', rawStr];
+      html.patch(elem, jml);
+      assert.equal(elem.outerHTML, '<div>foo</div>');
+    });
+
+    describe('element attributes', function() {
+      it('should be able to update element attributes', function() {
+        const elem = document.createElement('input');
+        const attrs = {
+          id: 'my-el-1',
+          name: 'my-best-el',
+          className: 'foo bar baz',
+          disabled: true,
+        };
+        const jml = ['', attrs];
+        html.patch(elem, jml);
+        assert.equal(elem.outerHTML, `<input id="${attrs.id}" name="${attrs.name}" class="${attrs.className}" disabled="">`);
+      });
+
+      it('should be able to update element styles through attributes', function() {
+        const elem = document.createElement('input');
+        // NOTE: JsonML style attribute must provide the literal inline CSS string value.
+        // An object of DOM style properties (e.g. marginTop) will not be applied as expected.
+        const attrs = {
+          style: 'color: blue; margin-top: 25px;',
+        };
+        const jml = ['', attrs];
+        html.patch(elem, jml);
+        assert.equal(elem.outerHTML, `<input style="${attrs.style}">`);
+      });
```

Comment:
I don't know why, but I initially expected the library to nicely handle a DOM style object. I guess that would be a nice feature in the future.